.github/workflows/acceptance-test.yml

```diff
@@ -0,0 +1,76 @@
+name: Acceptance test
+
+on:
+  workflow_call:
+    inputs:
+      secret:
+        required: true
+        type: string
+      secret-in-section:
+        required: true
+        type: string
+      multiline-secret:
+        required: true
+        type: string
+      export-env:
+        required: true
+        type: boolean
```

Comment:
`[note for reviewers]`
With this refactor, I'm considering that it would be a better idea to no longer pass this as part of the test, but rather have this reusable workflow do both i.e. run the tests for providing secrets as a step's output and as env vars. These would be run in parallel in a the defined matrix.

With that, we would increase our acceptance tests from 9 to 12: a set of 6 for references with names, and another 6 for references with IDs.

Reply:
Personally I like this PR as it already is. We could consider doing what you proposed as part of a separate PR, but given that a contributor relies on another PR which relies on this one, I'd say that we don't increase the scope of this refactor.